File: src/Stack/Stack.java
package Stack;

public class Stack<T> implements StackInterface<T> {
	private T[] array;
	private int top = -1;
	
	public Stack(int size) {
		this.array = (T[]) new Object[size];
	}
	
	@Override
	public void push(T element) throws StackOverflowException {
		if(element == null)
			return;
		if(!this.isFull()){
			this.array[++top] = element;
		}else{
			throw new StackOverflowException();
		}
	}

	@Override
	public T pop() throws StackUnderflowException {
		if(!this.isEmpty()){
			return array[this.top--];
		}else{
			throw new StackUnderflowException();
		}
	}

	@Override
	public T top() {
		if(!this.isEmpty()){
			return this.array[this.top];
		}else{
			return null;
		}
	}

	@Override
	public boolean isEmpty() {
		return this.top < 0;
	}

	@Override
	public boolean isFull() {
		return this.top == this.array.length - 1;
	}
	
	//If you want create a auto-resize stack
	private void resizeArray(){
		if(this.isFull()){
			T[] newArray = (T[]) new Object[this.array.length* 2];
			for(int i = 0 ; i < this.array.length;i++){
				newArray[i] = this.array[i];
			}
			this.array = newArray;
		}
	}
}

File: Python/Stack/Stack.py
'''
	Author: Ionésio Junior
'''
#coding:utf-8

'''
	Stack Implementation
'''
class Stack():

	def __init__(self,size = 10):
		self.__stackList = []
		self.__size = size
		self.__top = -1
		
	def push(self,element):
		if(element != None and  not(self.isFull())):
			self.__top = self.__top + 1
			self.__stackList.insert(self.__top,element)
		else:
			raise Exception("Stack is Full!!")
	def pop(self):
		if(not(self.isEmpty())):
			removedElement = self.__stackList[self.__top]
			del self.__stackList[self.__top]
			self.__top = self.__top - 1
			return removedElement
		else:
			raise Exception("Stack is empty!!")
	def top(self):
		if(not(self.isEmpty())):
			return self.__stackList[self.__top]
		else:
			return None
	def isEmpty(self):
		return (self.__top < 0)
	def isFull(self):
		return (self.__size - 1 ==  self.__top)
